filecheck/checker.ts

```diff
@@ -269,29 +269,60 @@ function canCheckFile(filePath: string) {
 async function resolveDirectory(file: string): Promise<string[]> {
   const stats = await fs.lstat(file);
   if (stats.isDirectory()) {
-    const api = new fdir()
-      .withErrors()
-      .withFullPaths()
-      .filter((filePath) => canCheckFile(filePath))
-      .crawl(file);
+    const api = new fdir().withErrors().withFullPaths().crawl(file);
     return api.withPromise() as Promise<PathsOutput>;
-  } else if (stats.isFile() && canCheckFile(file)) {
+  } else if (stats.isFile()) {
     return [file];
   } else {
     return [];
   }
 }
 
+function validatePath(filePath: string) {
+  // all characters must be lower case
+  const expectedPath = path.join(
+    path.basename(path.dirname(filePath)),
+    path.basename(filePath)
+  );
+  if (expectedPath !== expectedPath.toLowerCase()) {
+    throw new Error(
+      `Invalid path: ${expectedPath}. All characters must be lowercase.`
+    );
+  }
+
+  // file path should't contain banned characters: `(`, `)`
+  const bannedCharsRegExp = /\(|\)/;
+  if (bannedCharsRegExp.test(filePath)) {
+    throw new Error(
+      `Ivalid path: ${expectedPath}. File path must not include characters: '(', ')'`
+    );
+  }
+}
+
 export async function runChecker(
   filesAndDirectories: string[],
   options: CheckerOptions
 ) {
   const errors = [];
-
-  const files = (
+  let files = (
     await Promise.all(filesAndDirectories.map(resolveDirectory))
   ).flat();
 
+  // validate file paths
+  files = files.filter((fp) => !fp.includes(".json"));
```

Comment:
Rather than not validating `*.json` files at all, let's just allow them specifically to contain uppercase characters, inside `validatePath()`. Like this, we would still prevent them from containing parentheses (and whitespace, see above).

Reply:
done